packages/sdk/app-framework/src/plugins/next/plugin.ts

```diff
@@ -0,0 +1,264 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+/*
+- activation events in meta
+- plugins can contribute activation events
+- activate/deactivate functions (passed context to request capabilities from other plugins)
+- contribution point function contributing capabilities based on well-defined interfaces with with fully qualified names
+- allow plugin defs to be arrays of plugin defs
+- laziness needs to move from pluign def to the contribution point function
+*/
+
+import { untracked } from '@preact/signals-core';
+
+import { create } from '@dxos/live-object';
+import { log } from '@dxos/log';
+import { type MaybePromise } from '@dxos/util';
+
+/**
+ *
+ */
+export type PluginMeta = {
+  /**
+   * Globally unique ID.
+   *
+   * Expected to be in the form of a valid URL.
+   *
+   * @example dxos.org/plugin/example
+   */
+  id: string;
+
+  /**
+   * Human-readable name.
+   */
+  name?: string;
+
+  /**
+   * A grep-able symbol string which can be resolved to an icon asset by @ch-ui/icons, via @ch-ui/vite-plugin-icons.
+   */
+  icon?: string;
+
+  /**
+   * Short description of plugin functionality.
+   */
+  description?: string;
+
+  /**
+   * URL of home page.
+   */
+  homePage?: string;
+
+  /**
+   * URL of source code.
+   */
+  source?: string;
+
+  /**
+   * Tags to help categorize the plugin.
+   */
+  tags?: string[];
+};
+
+/**
+ *
+ */
+export type InterfaceDef<T> = {
+  // TODO(dima): Pattern borrowed from effect, they use symbols which I ommited for brewity.
+  _TypeId: {
```

Comment:
```ts
const InterfaceDefTypeId: unique symbol = Symbol.for(blah)

export type InterfaceDef<T> = {
  // TODO(dima): Pattern borrowed from effect, they use symbols which I ommited for brewity.
  [IntefaceDefTypeId]: {
  
  }
}
```

basically making it a `unique symbol` will hide it for intelisense in vscode

Reply:
btw, this pattern in type level programming is called a `Phantom Type`